### PR TITLE
Jetpack Cloud: fix the section property of the threat toggle event

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* The name of the section isn't always being tracked correctly. This is because we are using [`window.location.href` to get the current section](https://github.com/Automattic/wp-calypso/pull/42819/files#diff-c23e1d50bbefb88d4546c95f5d3d902aR98). This PR fixes this by using a state selector to fetch the current section.

#### Testing instructions

* Go to `http://jetpack.cloud.localhost:3000/` and select a site that has Scan (with at least one threat)
* Go to Scan -> Scanner 
* Click on the header of a threat item
* Verify the `calypso_jetpack_scan_threat_itemtoggle` event was tracked and that it included a property called `section` with `Scanner` as its value.
* Go to Scan -> History 
* Click on the header of a threat item
* Verify the `calypso_jetpack_scan_threat_itemtoggle` event was tracked and that it included a property called `section` with `History` as its value.

Fixes 1169345694087188-as-1178539912392265

#### Demo
**Scan -> Scanner**
image

**Scan -> History**
image

### CODE REVIEW

File: client/landing/jetpack-cloud/components/threat-item/index.tsx [hunk 1]
```diff
@@ -93,9 +95,10 @@ const ThreatItem: React.FC< Props > = ( {
 		[ threat ]
 	);
 
+	const currentRoute = useSelector( getCurrentRoute );
 	const onOpenTrackEvent = useTrackCallback( noop, 'calypso_jetpack_scan_threat_itemtoggle', {
 		threat_signature: threat.signature,
-		section: window.location.pathname.includes( '/scan/history' ) ? 'History' : 'Scanner',
+		section: currentRoute && currentRoute.includes( '/scan/history' ) ? 'History' : 'Scanner',
```
Comment (monsieur-z) [inline client/landing/jetpack-cloud/components/threat-item/index.tsx:101]: Should we define a section at all if `currentRoute` is not defined?

Comment (rcanepa) [reply client/landing/jetpack-cloud/components/threat-item/index.tsx:101]: > Probably I missed something, but I can't see the track events, no t.gif in my Network

To see an event, you need at least one threat in the History section. So you need to fix or ignore one to make it appear in that section. Then, clicking the threat will open it and track the event.

image



> Another thing, is this okay? The history seems empty.
>
> image

Yes. This is not great but it's have we have now when a user hasn't fixed/ignored any threat in the past. So, it's okay but we should improve it.

> And this is the API response:
> image
>
> is it okay?

Not sure about this one. It looks like the response we got when we had a fake endpoint. I'll investigate this one. Thanks @cleacos !

**Comment by rcanepa** [comment]
> Probably I missed something, but I can't see the track events, no t.gif in my Network

To see an event, you need at least one threat in the History section. So you need to fix or ignore one to make it appear in that section. Then, clicking the threat will open it and track the event.

image



> Another thing, is this okay? The history seems empty.
>
> image

Yes. This is not great but it's have we have now when a user hasn't fixed/ignored any threat in the past. So, it's okay but we should improve it.

> And this is the API response:
> image
>
> is it okay?

Not sure about this one. It looks like the response we got when we had a fake endpoint. I'll investigate this one. Thanks @cleacos !

File: client/landing/jetpack-cloud/components/threat-item/index.tsx (hunk 1 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/threat-item/index.tsx:101]: Good point. I'll remove the property if `currentRoute` is not defined.

**Comment by cleacos** [comment]
Now I can see the event `calypso_jetpack_scan_threat_itemtoggle`  with the property `section: History`

Why this event? because we are tracking toggle event (open and close), Why don't we just track the open event?

**Comment by cleacos** [comment]
still receiving the `lifetime_stats` with -1 values

**Comment by rcanepa** [comment]
> Now I can see the event `calypso_jetpack_scan_threat_itemtoggle` with the property `section: History`

Awesome!

> Why this event? because we are tracking toggle event (open and close), Why don't we just track the open event?

I don't have a great response. Eric said that toggle was good for our purposes and changing it to open/close would be a bit harder since the component that handles the `onClick` event is a very old component used in Calypso as well. I prefer not to touch it unless it's pretty necessary.

**Comment by rcanepa** [comment]
> still receiving the `lifetime_stats` with -1 values

Yup. I investigated this and we are seeing that because that was a fake response we used when we started to work on Scan. I created a card to surface this issue and decide what to do about it 1169345694087188-as-1178763037766308.

File: client/landing/jetpack-cloud/components/threat-item/index.tsx (hunk 1 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/threat-item/index.tsx]: I updated this @monsieur-z . Whenever you have time, I would love to have your eyes on this again :)

**Pull request merged**